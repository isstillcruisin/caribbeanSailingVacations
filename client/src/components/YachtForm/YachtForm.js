import React from 'react'
import ImageUploader from '../ImageUploader'
import Container from 'react-bootstrap/Container'
import Alert from '../Alert'
import PropTypes from 'prop-types'
import Form from 'react-bootstrap/Form'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Card from 'react-bootstrap/Card'

const YachtForm = props => (
  <Container>
    <Alert alert={props.alert} />      
    <Form>
      <Form.Row controlId="form-boat-name">
        <Col xs="2">
          <Form.Label>Boat Name</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='text' 
            value={props.boatName}
            placeholder='name of the boat...' 
            name='boatName'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId="form-crew-bio">
        <Col xs="2">
          <Form.Label>Crew Bio</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            as='textarea'
            rows={4} 
            value={props.crewBio}
            placeholder='about the crew...' 
            name='crewBio'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
          <Form.Row controlId="form-max-passengers">
        <Col xs="2">
          <Form.Label>Max Passengers</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='number'
            value={props.maxPassengers}
            placeholder='maximum number of passengers on the yacht' 
            name='maxPassengers'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId="form-year">
        <Col xs="2">
          <Form.Label>Year Made</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='number'
            value={props.year}
            placeholder='year the yacht was constructed' 
            name='year'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId="form-manufacture">
        <Col xs="2">
          <Form.Label>Manufacture of Yacht</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='text'
            value={props.manufacture}
            placeholder='Boat size, etc...' 
            name='manufacture'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId="form-price-per-week">
        <Col xs="2">
          <Form.Label>Price Per Week</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='number'
            value={props.pricePerWeek}
            placeholder='price per week ($)' 
            name='pricePerWeek'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId="form-cya-id">
        <Col xs="2">
          <Form.Label>Central Yacht Agent ID</Form.Label>
        </Col>
        <Col>
          <Form.Control 
            type='text'
            value={props.cyaId}
            placeholder='Central Yacht Agent ID' 
            name='cyaId'
            onChange={props.handleInputChange}
            required
          />
        </Col>
      </Form.Row>
      <Form.Row controlId='form-images'>
        <Col xs='2'>
          <Form.Label>Boat Images</Form.Label>
        </Col>
        <Col xs='10'>
          <ImageUploader
            imgs={props.imgs}
            setUrls={props.handleSetUrls}
            placeholder='Drop a file here, or click to select a file to upload.'
            multiple={true}
            showDropZone={true}
          />
        </Col>
      </Form.Row>
      <div className="pull-right">
        <button
          onClick={props.handleFormSubmit}
          type="submit"
          className="btn btn-lg btn-danger"
        >
          Save
        </button>
      </div>
      { props.cyaId && (
        <div className='mt-2'>
          <a href={`https://www.centralyachtagent.com/yachtadmin/showyachts.php?id=${props.cyaId}&disptype=1&uid=1892&unit=1&stat=1`} target='_blank' rel="noopener noreferrer">
            CYA E-Brochure (opens in new tab)
          </a>
        </div>
      )}
    </Form>
  </Container>
)

YachtForm.propTypes = {
  cyaId: PropTypes.number,
  alert: PropTypes.string,
  boatName: PropTypes.string,
  handleInputChange: PropTypes.func,
  crewBio: PropTypes.string,
  maxPassengers: PropTypes.number,
  year: PropTypes.number,
  manufacture: PropTypes.string,
  pricePerWeek: PropTypes.number,
  imgs: PropTypes.arrayOf(PropTypes.string),
  handleSetUrls: PropTypes.func,
  handleFormSubmit: PropTypes.func,
}

export default YachtForm
