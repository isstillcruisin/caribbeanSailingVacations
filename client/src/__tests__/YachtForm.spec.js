import React from 'react'
import { create } from 'react-test-renderer'
import YachtForm from '../components/YachtForm'

describe('YachtForm component', () => {
  test('Matches the snapshot', () => {
    const form = create(<YachtForm 
      boatName='test boat'
      crewBio='test bio'
      maxPassengers={10}
      year={1999}
      manufacture='2000 feet test manufacture'
      imgs={[]}
    />)
    expect(form.toJSON()).toMatchSnapshot()
  })
})