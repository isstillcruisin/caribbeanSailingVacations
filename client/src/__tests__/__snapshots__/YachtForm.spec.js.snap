// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YachtForm component Matches the snapshot 1`] = `
<div
  className="container"
>
  
  <form
    className=""
  >
    <div
      className="form-row"
      controlId="form-boat-name"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Boat Name
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="boatName"
          placeholder="name of the boat..."
          required={true}
          type="text"
          value="test boat"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-crew-bio"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Crew Bio
        </label>
      </div>
      <div
        className="col"
      >
        <textarea
          className="form-control"
          name="crewBio"
          placeholder="about the crew..."
          required={true}
          rows={4}
          value="test bio"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-max-passengers"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Max Passengers
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="maxPassengers"
          placeholder="maximum number of passengers on the yacht"
          required={true}
          type="number"
          value={10}
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-year"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Year Made
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="year"
          placeholder="year the yacht was constructed"
          required={true}
          type="number"
          value={1999}
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-manufacture"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Manufacture of Yacht
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="manufacture"
          placeholder="Boat size, etc..."
          required={true}
          type="text"
          value="2000 feet test manufacture"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-price-per-week"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Price Per Week
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="pricePerWeek"
          placeholder="price per week ($)"
          required={true}
          type="number"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-cya-id"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Central Yacht Agent ID
        </label>
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          name="cyaId"
          placeholder="Central Yacht Agent ID"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="form-row"
      controlId="form-images"
    >
      <div
        className="col-2"
      >
        <label
          className="form-label"
        >
          Boat Images
        </label>
      </div>
      <div
        className="col-10"
      >
        <div
          className="boat-image-dropzone"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <input
            accept="image/*"
            autoComplete="off"
            multiple={true}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            tabIndex={-1}
            type="file"
          />
          <p>
            Drop a file here, or click to select a file to upload.
          </p>
        </div>
        <div>
          <div
            className="image-thumbnails"
          />
        </div>
      </div>
    </div>
    <div
      className="pull-right"
    >
      <button
        className="btn btn-lg btn-danger"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
